x/distribution/keeper/hooks.go

```diff
@@ -77,6 +78,14 @@ func (k Keeper) onDelegationCreated(ctx sdk.Context, delAddr sdk.AccAddress,
 func (k Keeper) onDelegationSharesModified(ctx sdk.Context, delAddr sdk.AccAddress,
 	valAddr sdk.ValAddress) {
 
+	if bytes.Equal(delAddr.Bytes(), valAddr.Bytes()) {
+		// On updates to a self bond/unbond, we must update the validator's dist
+		// info without withdrawing any rewards.
+		k.UpdateValidatorDistInfoFromPool(ctx, valAddr)
```

Comment:
Also if you look at what's happening within onValidatorModified you'll see we skip running this at height of 0 - don't see why we wouldn't replicate this for the new function too

Reply:
Ditto: I don't think there is a need now as I just moved the new call into `onValidatorModified`.